packages/astro/src/core/sync/write-files.ts

```diff
@@ -74,3 +80,36 @@ async function setUpEnvTs(settings: AstroSettings, fs: typeof fsMod, logger: Log
 		logger.info('types', `Added ${bold(envTsPathRelativetoRoot)} type declarations`);
 	}
 }
+
+async function setupTsconfig(settings: AstroSettings, fs: typeof fsMod, logger: Logger) {
+	const tsconfig = settings.config.experimental.tsconfig!;
```

Comment:
Please add a comment explaining why the bang operator is safe here. I presume because the function is always called when the experimental flag is on.